### PR TITLE
Be more lenient with non-standard pre-release versions (alpha12 -> alpha.12)

### LINKED ISSUE #9: Pre-releases are not treated the same than in Composer
see https://github.com/FriendsOfPHP/security-advisories/issues/511 for the initial report.

It looks like the comparison is done using a stricter semver implementation than composer which does require a `.` to separate the alphabetical identifier from the following pre-release version. Composer makes that `.` optional, treating `2.0.0-alpha11` as it it was `2.0.0-alpha.11` in semver.

### PR DESCRIPTION
**Pull request by pableu**
This fixes #9.

### CODE REVIEW

File: security/version.go [hunk 1]
```diff
@@ -20,6 +21,11 @@ func (v *Version) UnmarshalJSON(b []byte) error {
 	if err := json.Unmarshal(b, &tmp); err != nil {
 		return err
 	}
+
+	// be more lenient with pre-release versions, convert "2.0.0-alpha12" to "2.0.0-alpha.12"
+	re := regexp.MustCompile(`(alpha|beta)(\d+)$`)
```
Comment (stof) [inline security/version.go:26]: what about `rc` ?

Comment (pableu) [reply security/version.go:26]: You're right. If a understand the [semver-specs](https://semver.org/), the pre-release-identifier could be made up of arbitrary alphanumerics, i.e. something like "1.0.0-**xyz**.9" would be valid. 

But I think that's a bit risky. We simply want to work around other packages that don't do the dot separator properly, not re-invent a complete semver-library ;-)

@fabpot should I make a new PR to add `rc` to the regexp?

Comment (stof) [reply security/version.go:26]: but composer itself supports `alpha`, `beta` and `rc` explicitly (and case-insensitively, which is also missing here AFAICT)

Comment (pableu) [reply security/version.go:26]: Okay perfect, thanks. Then I'll do the same here :-)

Comment (pableu) [reply security/version.go:26]: See #28.

**Pull request merged**